### PR TITLE
Added mergify configuration

### PR DESCRIPTION
**Pull request by azidar**
To start, this will allow mergify to do stuff only if the "Please Merge" label is present.

When merging, it will queue up the "Please Merge" PRs, and for each, merge when updating the PR to master, and the squash-and-merge when upstreaming the PR.

We can think about (1) rebasing onto master or (2) making mergify the default (not need the "Please Merge" label) for merging things that are approved.

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      dismiss_reviews:
+
+pull_request_rules:
+  - name: automatic squash-and-merge on CI success and review
+    conditions:
+      - status-success=continuous-integration/travis-ci/pr
+      - "#approved-reviews-by>=1"
+      - "#review-requested=0"
+      - "#changes-requested-reviews-by=0"
+      - "#commented-reviews-by=0"
```
Comment (seldridge) [inline .mergify.yml]: Possibly remove this one. It's not clear to me if this should be a blocker.

**Comment by ucbjrl** [comment]
Is it up to the original submitter to delete the old PR branch?

**Comment by edwardcwang** [comment]
Seems like mergify does NOT delete the stale PR branch, so someone has to do it manually

**Comment by seldridge** [comment]
Not enabled, but I think is supported: https://doc.mergify.io/actions.html#delete-head-branch

**Pull request merged**